Dockerfile

```diff
@@ -13,25 +13,23 @@ FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-alpine AS golatest
 
 FROM golatest AS gobase
 COPY --from=xx / /
-RUN apk add --no-cache file git
+RUN apk add --no-cache file git make
 ENV GOFLAGS=-mod=vendor
 ENV CGO_ENABLED=0
 WORKDIR /src
 
 FROM gobase AS buildx-version
 RUN --mount=target=. \
-  PKG=github.com/docker/buildx VERSION=$(git describe --match 'v[0-9]*' --dirty='.m' --always --tags) REVISION=$(git rev-parse HEAD)$(if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi); \
-  echo "-X ${PKG}/version.Version=${VERSION} -X ${PKG}/version.Revision=${REVISION} -X ${PKG}/version.Package=${PKG}" | tee /tmp/.ldflags; \
-  echo -n "${VERSION}" | tee /tmp/.version;
+  echo -n "$(make ldflags)" | tee /tmp/.ldflags ; echo -n "$(make version)" | tee /tmp/.version
 
 FROM gobase AS buildx-build
-ARG LDFLAGS="-w -s"
 ARG TARGETPLATFORM
+RUN xx-go --wrap
 RUN --mount=type=bind,target=. \
   --mount=type=cache,target=/root/.cache \
   --mount=type=cache,target=/go/pkg/mod \
   --mount=type=bind,source=/tmp/.ldflags,target=/tmp/.ldflags,from=buildx-version \
-  set -x; xx-go build -ldflags "$(cat /tmp/.ldflags) ${LDFLAGS}" -o /usr/bin/buildx ./cmd/buildx && \
+  make build CGO_ENABLED=$CGO_ENABLED LDFLAGS="$(cat /tmp/.ldflags)" DESTDIR=/usr/bin BINNAME=buildx && \
```

Comment:
Why this change? How should the user know if they need a specific environment for (some of) the make targets. Seems quite messy to combine these. Same for the `buildx-version`

Reply:
I changed the implementation so it does not require `make` in the Dockerfile. This commit is to avoid drifting with build opts when buildx is packaged in https://github.com/docker/docker-ce-packaging/blob/5683c6bdcabf000cec58677ebb5f411f5410512a/deb/common/rules#L22. Let me know if you prefer to move it in another PR.